mllib/src/main/scala/org/apache/spark/ml/feature/LSH.scala

```diff
@@ -112,9 +113,7 @@ private[ml] abstract class LSHModel[T <: LSHModel[T]]
       numNearestNeighbors: Int,
       singleProbe: Boolean,
       distCol: String): Dataset[_] = {
-    val count = dataset.count()
-    require(numNearestNeighbors > 0 && numNearestNeighbors <= count, "The number of" +
-      " nearest neighbors cannot be less than 1 or greater than the number of elements in dataset")
+    require(numNearestNeighbors > 0, "The number of nearest neighbors cannot be less than 1")
```

Comment:
As discussed in https://github.com/apache/spark/pull/26415,

"@param numNearestNeighbors The maximum number of nearest neighbors."
It implies it could return fewer items.